### PR TITLE
Create mergeSort.py

### PR DESCRIPTION
**Pull request by Varchasavkr**
This is a code for mergeSort in python

def merge(arr, l, m, r):
	n1 = m - l + 1
	n2 = r - m

	# create temp arrays
	L = [0] * (n1)
	R = [0] * (n2)

	# Copy data to temp arrays L[] and R[]
	for i in range(0, n1):
		L[i] = arr[l + i]

	for j in range(0, n2):
		R[j] = arr[m + 1 + j]

	i = 0	 # Initial index of first subarray
	j = 0	 # Initial index of second subarray
	k = l	 # Initial index of merged subarray

	while i < n1 and j < n2:
		if L[i] <= R[j]:
			arr[k] = L[i]
			i += 1
		else:
			arr[k] = R[j]
			j += 1
		k += 1

	
	while i < n1:
		arr[k] = L[i]
		i += 1
		k += 1

	while j < n2:
		arr[k] = R[j]
		j += 1
		k += 1

# l is for left index and r is right index of the
# sub-array of arr to be sorted

def mergeSort(arr, l, r):
	if l < r:

		# Same as (l+r)//2, but avoids overflow for
		# large l and h
		m = l+(r-l)//2

		# Sort first and second halves
		mergeSort(arr, l, m)
		mergeSort(arr, m+1, r)
		merge(arr, l, m, r)

# Driver code to test above
arr = [12, 11, 13, 5, 6, 7]
n = len(arr)
print("Given array is")
for i in range(n):
	print("%d" % arr[i],end=" ")

mergeSort(arr, 0, n-1)
print("\n\nSorted array is")
for i in range(n):
	print("%d" % arr[i],end=" ")
	
	this is the code added 
	
	kindly merge this under #HacktoberFest2024 and #gssoc-ext #level2

### CODE REVIEW

**Comment by Varchasavkr** [comment]
@anantcodes 
kindly merge this PR 
can you attach these labels
#hacktoberfest-accepted #gssoc-ext #level2

File: mergeSort.py [hunk 1]
```diff
@@ -0,0 +1,74 @@
+
```
Comment (anantcodes) [inline mergeSort.py:1]: Please remove extraneous spaces.

File: mergeSort.py [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+	# are any
+	while j < n2:
+		arr[k] = R[j]
+		j += 1
+		k += 1
+
+# l is for left index and r is right index of the
+# sub-array of arr to be sorted
+
+
```
Comment (anantcodes) [inline mergeSort.py:48]: Here as well

File: mergeSort.py [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+		# Same as (l+r)//2, but avoids overflow for
+		# large l and h
+		m = l+(r-l)//2
+
+		# Sort first and second halves
+		mergeSort(arr, l, m)
+		mergeSort(arr, m+1, r)
+		merge(arr, l, m, r)
+
+
```
Comment (anantcodes) [inline mergeSort.py:61]: Here as well

File: mergeSort.py [hunk 4]
```diff
@@ -0,0 +1,74 @@
...
+print("Given array is")
+for i in range(n):
+	print("%d" % arr[i],end=" ")
+
+mergeSort(arr, 0, n-1)
+print("\n\nSorted array is")
+for i in range(n):
+	print("%d" % arr[i],end=" ")
+
+
```
Comment (anantcodes) [inline mergeSort.py:74]: Here as well

**Review by anantcodes — CHANGES_REQUESTED** [review]
Please make these changes. Thanks!

**Comment by Varchasavkr** [comment]
@anantcodes i can removed all extra spaces and made another commit with updated mergesort.py name
kindly merge it and attach label of #hacktoberfest-accepted and #gssoc-ext and #level3

**Review by anantcodes — APPROVED** [review]
Lgtm

**Comment by Varchasavkr** [comment]
@anantcodes can please add other label as mentioned above also 
Humbly asking 
Thankyou

**Comment by anantcodes** [comment]
> @anantcodes can please add other label as mentioned above also Humbly asking Thankyou

Hey @Varchasavkr , this repository is only open for hacktoberfest contributions for building a dynamic algorithm repository for industrial purposes. I'll not be able to add other events here.

**Pull request merged**